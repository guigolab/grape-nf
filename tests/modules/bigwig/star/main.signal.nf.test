nextflow_process {

    name "Test Process signal"
    script "modules/bigwig/star/main.nf"
    process "signal"

    tag "module"

    test("Should produce signal files using STAR") {

        setup {
            run("fastaIndex") {
                script = "modules/fastaIndex/samtools/main.nf"
                process {
                    """
                    input[0] = "${baseDir}/data/genome.fa.gz"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = fastaIndex.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE"]
                ])
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                def signalDir = get(0)[0][2]
                def meta = get(0)[0] - signalDir
                def signalFiles = path(signalDir)
                                    .list()
                                    .findAll({
                                        "${it.fileName}" != "Log.out"
                                    })
                assert signalFiles.size() == 4
                assert snapshot(
                    meta,
                    signalFiles
                ).match()
            }
        }

    }

}
