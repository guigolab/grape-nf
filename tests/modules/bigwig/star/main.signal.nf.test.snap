{
    "Should produce signal files with STAR": {
        "content": [
            [
                "sample3",
                "test3",
                "bedGraph",
                "SignalBedgraphs",
                true,
                "MATE2_SENSE"
            ],
            [
                "Signal.Unique.str1.out.bg:md5,68d27d2751d4ad7d6cddedcf2111cc87",
                "Signal.Unique.str2.out.bg:md5,47008cc228a8dd00167430c22edb8033",
                "Signal.UniqueMultiple.str1.out.bg:md5,93302fecb5013e7e6ad5dc21b0811d73",
                "Signal.UniqueMultiple.str2.out.bg:md5,d4c1161f9d27d2492efe6b24348a3501"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-25T12:33:44.577831"
    }
}